docker/docker_image_dest.go

```diff
@@ -363,6 +365,15 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 			remoteName:   reference.Path(candidateRepo),
 			actions:      "pull",
 		}
+		compressionOperation, compressionAlgorithm, err := blobinfocache.OperationAndAlgorithmForCompressor(candidate.CompressorName)
+		if err != nil {
+			logrus.Debugf("... Failed: %v", err)
+			continue
+		}
+		if !impl.BlobMatchesRequiredCompression(options, compressionAlgorithm) {
+			logrus.Debugf("Ignoring candidate blob %s as reuse candidate since compression %s was requested and blob has compression %s", candidate.Digest.String(), options.RequiredCompression.Name(), compressionAlgorithm.Name())
```

Comment:
This also needs to not crash on `compressionAlgorithm == nil`.

(Also compare the “Trying to reuse” code; including `candidateRepo.Name()` seems useful — especially if the code is moved so that the “trying” logs aren’t emitted at all.)

(Absolutely non-blocking: making this a bit shorter would help on narrow terminals / log viewers: “Not using …, compression mismatch (%s vs. %s)”. The text should make sense but it can probably be assumed the reader can refer to the related source code, so it doesn’t need to be quite self-documenting.)